test/client/fixtures/sandbox/message-test.js

```diff
@@ -14,6 +14,24 @@ QUnit.testDone(function () {
     iframeSandbox.off(iframeSandbox.IFRAME_READY_TO_INIT_EVENT, initIFrameTestHandler);
 });
 
+asyncTest('onmessage event (handler has "object" type) (GH-133)', function () {
+    var testMessage       = 'test';
+    var testPropertyValue = 'propertyValue';
+
+    var eventHandlerObject = {
+        propertyName: testPropertyValue,
+        handleEvent:  function (e) {
+            strictEqual(e.data, testMessage);
+            strictEqual(this.propertyName, testPropertyValue);
```

Comment:
`strictEqual(this.propertyName, testPropertyValue);`

replace to:

 `strictEqual(this, eventHandlerObject);`